org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/validation/DomainmodelValidator.xtend

```diff
@@ -23,25 +28,25 @@ import org.eclipse.xtext.validation.ValidationMessageAcceptor
 class DomainmodelValidator extends AbstractDomainmodelValidator {
 
 	@Check def void checkTypeNameStartsWithCapital(Entity entity) {
-		if (!Character::isUpperCase(entity.getName().charAt(0))) {
-			warning("Name should start with a capital", DomainmodelPackage.Literals::ABSTRACT_ELEMENT__NAME,
-				ValidationMessageAcceptor::INSIGNIFICANT_INDEX, IssueCodes::INVALID_TYPE_NAME, entity.getName())
+		if (!entity.name.charAt(0).isUpperCase) {
+			warning("Name should start with a capital", ABSTRACT_ELEMENT__NAME,
+				ValidationMessageAcceptor.INSIGNIFICANT_INDEX, INVALID_TYPE_NAME, entity.name)
 		}
 	}
 
 	@Check def void checkFeatureNameStartsWithLowercase(Feature feature) {
-		if (!Character::isLowerCase(feature.getName().charAt(0))) {
-			warning("Name should start with a lowercase", DomainmodelPackage.Literals::FEATURE__NAME,
-				ValidationMessageAcceptor::INSIGNIFICANT_INDEX, IssueCodes::INVALID_FEATURE_NAME, feature.getName())
+		if (!feature.name.charAt(0).isLowerCase) {
+			warning("Name should start with a lowercase", FEATURE__NAME,
+				ValidationMessageAcceptor.INSIGNIFICANT_INDEX, INVALID_FEATURE_NAME, feature.name)
 		}
 	}
 
 	@Check def void checkPackage(PackageDeclaration packages) {
-		if (Strings::isEmpty(packages.getName())) {
-			error("Name cannot be empty", DomainmodelPackage.Literals::ABSTRACT_ELEMENT__NAME)
+		if (Strings.isEmpty(packages.name)) {
```

Comment:
Can be shorted to `packages.name.empty`

Reply:
The `packages.name.empty` call will be bound to the java.lang.String.isEmpty call, so it will be different from the `org.eclipse.xtext.util.Strings.isEmpty` call when handling `null` values. So I will leave that method call as it is to be safe.

Reply:
How about `packages.name?.empty` then?

Reply:
That one results in the following warning:
![warning](https://user-images.githubusercontent.com/20393472/65878510-894e9300-e38e-11e9-93eb-f399147adbca.png)